datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/aspect/AspectMapper.java

```diff
@@ -1,26 +1,106 @@
+import com.linkedin.datahub.graphql.types.glossary.mappers.GlossaryTermsMapper;
 import com.linkedin.datahub.graphql.types.mappers.ModelMapper;
+import com.linkedin.datahub.graphql.types.tag.mappers.GlobalTagsMapper;
+import com.linkedin.dataset.DatasetProperties;
+import com.linkedin.dataset.EditableDatasetProperties;
+import com.linkedin.entity.EntityResponse;
 import com.linkedin.entity.EnvelopedAspect;
-import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.key.DatasetKey;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.schema.EditableSchemaMetadata;
+import com.linkedin.util.Pair;
 import javax.annotation.Nonnull;
 
+import static com.linkedin.metadata.Constants.*;
 
-public class AspectMapper implements ModelMapper<EnvelopedAspect, Aspect> {
+
+public class AspectMapper implements ModelMapper<Pair<VersionedAspectKey, EntityResponse>, Aspect> {
 
   public static final AspectMapper INSTANCE = new AspectMapper();
 
-  public static Aspect map(@Nonnull final EnvelopedAspect aspect) {
-    return INSTANCE.apply(aspect);
+  public static Aspect map(@Nonnull final Pair<VersionedAspectKey, EntityResponse> responsePair) {
+    return INSTANCE.apply(responsePair);
   }
 
   @Override
-  public Aspect apply(@Nonnull final EnvelopedAspect aspect) {
-    if (Constants.SCHEMA_METADATA_ASPECT_NAME.equals(aspect.getName())) {
-      return SchemaMetadataMapper.map(aspect);
+  public Aspect apply(@Nonnull final Pair<VersionedAspectKey, EntityResponse> responsePair) {
+    EntityResponse response = responsePair.getSecond();
+    EnvelopedAspect aspect = response.getAspects().get(responsePair.getFirst().getAspectName());
+    switch (aspect.getName()) {
+      case SCHEMA_METADATA_ASPECT_NAME:
+        return SchemaMetadataMapper.map(aspect);
+      case GLOBAL_TAGS_ASPECT_NAME:
+        com.linkedin.datahub.graphql.generated.GlobalTags globalTags = GlobalTagsMapper
+            .map(new GlobalTags(aspect.getValue().data()));
+        globalTags.setVersion(aspect.getVersion());
+        return globalTags;
+      case GLOSSARY_TERMS_ASPECT_NAME:
+        com.linkedin.datahub.graphql.generated.GlossaryTerms glossaryTerms = GlossaryTermsMapper
+            .map(new GlossaryTerms(aspect.getValue().data()));
+        glossaryTerms.setVersion(aspect.getVersion());
+        return glossaryTerms;
+      case DATASET_PROPERTIES_ASPECT_NAME:
+        return mapDatasetProperties(response, aspect);
+      case EDITABLE_DATASET_PROPERTIES_ASPECT_NAME:
+        DatasetEditableProperties datasetEditableProperties = new DatasetEditableProperties();
+        EditableDatasetProperties editableDatasetProperties = new EditableDatasetProperties(aspect.getValue().data());
+        datasetEditableProperties.setDescription(editableDatasetProperties.getDescription());
+        datasetEditableProperties.setVersion(aspect.getVersion());
+        return datasetEditableProperties;
+      case EDITABLE_SCHEMA_METADATA_ASPECT_NAME:
```

Comment:
im a bit concerned about encoding business logic in each one of the switch statements.. is there a way we can do this in a bit more readable format?

Reply:
The setVersions? I can golf it to feed in the functions and mapper types like I did with the MappingUtil class. For the non-trivial one (dataset properties) I pushed it out to a separate function already. Currently all the Mappers take their specific type instead of an EnvelopedAspect and I didn't think it was worth converting all of them for a single setter call.

Or is this just the EditableProperties one which is a couple of extra lines causing the issue? I can pull that out to a separate function